### ISSUE TITLE
[new-backend] Release

### ISSUE THREAD

**Issue by kodebach** [ElektraInitiative/libelektra#4483]
This issue is a overview and discussion issue to document what conditions must be met before we can merge `new-backend` into `master` (i.e. merge #4187).

- [ ] By the end of this year `master` must be in a usable and in a well-tested state.
- [ ] Docs for new-backend should be complete enough for newcomers to understand (FLOSS T3)
- [ ] Most of the tests should be reenabled.
- [ ] New tests for data structure written.
- [ ] There should be some way of migrating (e.g. via a script) `system:/elektra/mountpoints` to the new structure (@kodebach)
- [ ] mmap cache fixed again (@mpranj)
- [ ] Docs must not contain incorrect information

@markus2330 Please extend this list.

**Comment by markus2330** [comment]
Thank you for your initiative! :rocket: :heart_decoration:

I extended/updated the list above.

I don't think that all of this is needed for merging, instead it is needed before doing a release after merging the new-backend branch. I updated the title.

I created a new issue for what I think is needed for merging:  #4485

**Comment by kodebach** [comment]
> mmap fixed again

I guess you mean the cache should work? Because `mmapstorage` itself should already work.

**Comment by markus2330** [comment]
@kodebach wrote in #4436:

> I also didn't say that we should wait until February to merge new-backend.

If not now and not in Februar it means that it needs to happen in the middle of the term, which is difficult/impossible to communicate to the FLOSS participants, who want to work on their PRs with as little disturbances as possible.

> That would be a good thing. Since it would mean their work is finished, i.e. new-backend would be much closer to merge.

This wish is contrary to my experiences in the past. Only if work is actually used by someone else (which requires merged in master and ideally released and packaged) important problems that are essential for real-world use get discovered (e.g. #4337). When people are not available by then, real-world adoption gets unlikely. So putting the code somewhere in a branch that is not merged before finishing the thesis is practically a disaster.

**Comment by markus2330** [comment]
I updated the list above.
